### DISCUSSION TITLE
WebForms turbo mode compatibility issues ()

### DISCUSSION THREAD

**Discussion by rebimol** [breezefront/community#305] [1. Help]
This is Vladimir from MageMe. We have been developing the MageMe WebForms plugin and released the WebForms Breeze compatibility module over a year ago. 

We also developed an add-on for the WebForms plugin which lets you replace the standard custom options of the product with the form built with our form builder on a catalog page (also has Breeze compatibility module). 

The problem is that once you switch on the turbo mode some important JavaScript libraries are loaded randomly (not always) via the requirejs script like the conditional logic script and form validation on catalog pages where the form is integrated.

One of our clients has it working on staging site here (click on Gloves to see conditional logic in action):

https://staging.leathercollection.com/custom-made-leather-suit-japan-flag.html

But has problems with another store view on the same site (click on Gloves , logic is not working, js error in console):

https://stagingms.leathercollection.com/tyro-for-beginners.html

The bundles are randomly generated depending on which page of the site you are after refreshing the cache.

Is there any trick to make sure the requirejs is 100% loading the library in turbo mode?

We have made a breeze_default.xml file and it works absolutely fine without the turbo mode enabled. But once turbo is on, everything gets funky.

Thank you for your time. Looking forward to your response.

**Comment by vovayatsyuk** [comment]
Hello!

The [first site](https://staging.leathercollection.com/custom-made-leather-suit-japan-flag.html) has a js error because of the missing `MageMe_WebFormsProductOptions/js/submit` info inside `<script type="breeze/dynamic-js">` tag. 

This error stops all other inline js `require` calls.

Do you have any idea why `MageMe_WebFormsProductOptions/js/submit` info is added on the second site, but not on the first?

**Reply by rebimol** [reply]
It is randomly there. If I refresh the cache and start browsing the site from home page it doesn't get there. If I start browsing right from the catalog page it is there. Initially those 2 links were working oppositely, the first one was working and the second one wasn't.

Is there any way to force the script to get into the `<script type="breeze/dynamic-js">` ?  I suspect there could be a sequence thing, the breeze/dynamic-js could be formed before our module creates its output.

**Reply by vovayatsyuk** [reply]
Is it possible that `MageMe_WebFormsProductOptions/js/submit` is not declared in `breeze_default` but in another layout update file?

**Reply by rebimol** [reply]
It is in breeze_catalog_product_view.xml . Should it be always breeze_default?

**Reply by vovayatsyuk** [reply]
Yes, you need to move it to breeze_default.xml. Otherwise, breeze js config and bundles will be fragmented.

**Reply by rebimol** [reply]
This seem to have worked. Thank you for your help!
